sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java

```diff
@@ -249,9 +252,18 @@ private static void readDefaultOptionValues(
    * @param options the SentryOptions
    */
   private static void initializeCacheDirs(
-      final @NotNull Context context, final @NotNull SentryOptions options) {
+      final @NotNull Context context, final @NotNull SentryAndroidOptions options) {
     final File cacheDir = new File(context.getCacheDir(), "sentry");
+    final File profilingTracesDir = new File(cacheDir, "profiling_traces");
     options.setCacheDirPath(cacheDir.getAbsolutePath());
+    options.setProfilingTracesDirPath(profilingTracesDir.getAbsolutePath());
+
+    if (options.isProfilingEnabled()) {
+      // Method trace files are normally deleted at the end of traces, but if that fails for some
+      // reason we try to clear any old files here.
+      FileUtils.deleteRecursively(profilingTracesDir);
```

Comment:
This might slow down the start-up of Apps, is it always 1 file, or could be N?

Reply:
Also, a transaction can be started (and profiled) at any moment, even during the start-up, so some operation will still be needed to do on the main thread

Reply:
I see.

> i may add a check to avoid the deletion of an empty folder

That sounds an improvement, I'd do it.

Reply:
We could also do that in the background, using the `executorService`, but that would require atomic operations since a transaction might start while you are still deleting, your call, happy with the improvement above.
Because the empty folder will be deleted and recreated every time, so we spare some bits for 99% of the use cases.

Reply:
another thing i could do is to call `listFiles()` in the main thread and then delete only them in the background: since their names are uuids there's no problem on possible conflicts with new profiles

Reply:
Yeah, sounds better, so let's do this.
Don't delete the folder, and recreate it, if empty.
Just delete the inner files, if any.
List the files within the folder and delete them in the background.